### PR TITLE
TF Support for Firewall Rules(Resource,DataSource,Testcases and Docs)

### PR DESCRIPTION
**Pull request by shivaleelac**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Argument reference
+The following arguments are supported:
+
+- `cis_id` - (Required, String) The ID of the CIS service instance.
+- `domain_id` - (Required, String) The ID of the domain.
+
+## Attributes Reference
+In addition to all arguments above, the following attributes are exported:
+- `id` - (String) The Firewall rules ID. It is a combination of <`firewall_rule_id`>,<`domain_id`>,<`cis_id`> attributes concatenated with ":"
+- `cis_firewallrules_list` - (List of Firewall Rules)
```
Comment (kavya498) [inline website/docs/d/cis_firewall_rules.html.markdown:30]: In schema it is `firewall_rules` not `cis_firewallrules_list`
Please cross check all the attributes with schema

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - `paused` - (Boolean). Whether this firewall rules is currently disabled.
+  - `description` - (String) The information about this firewall rules to help identify the purpose of it.
+  - `action` - (String). Create a firewall rules using these actions `log`, `allow`, `challenge`, `js_challenge`, `block` 
+The firewall action to perform, "log" action is only available for enterprise plan instances.
+   
+   Nested scheme for `filter_id`:
```
Comment (kavya498) [inline website/docs/d/cis_firewall_rules.html.markdown:36]: `filter` is a typeMap.. description ofor filter is missing..
filter_id attribute is not a complex attribute in schema

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+```
+
+## Argument reference
+The following arguments are supported:
+
+- `cis_id` - (Required, String) The ID of the CIS service instance.
+- `domain_id` - (Required, String) The ID of the domain.
+
+## Attributes Reference
```
Comment (geethasathyamurthy) [inline website/docs/d/cis_firewall_rules.html.markdown:27]: R in Reference should be as **reference**

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+- `action` - (String) Create a firewall rules using these actions `log`, `allow`, `challenge`, `js_challenge`, `block` The firewall action to perform, "log" action is only available for enterprise plan instances.
```
Comment (geethasathyamurthy) [inline website/docs/d/cis_firewall_rules.html.markdown:30]: Create firewall rules by using `log`, `allow`, `challenge`, `js_challenge`, `block` actions. The firewall action to perform, `log` action is only available for the Enterprise plan instances.

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+- `description` - (String) The information about this firewall rules to help identify the purpose of it.
```
Comment (geethasathyamurthy) [inline website/docs/d/cis_firewall_rules.html.markdown:31]: The information about these firewall rules helps identify the purpose.

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+- `firewall_rules` - (List of Firewall Rules)
+- `filter` - (Map) An existing filter which contains expression, paused and description.
+
+  Nested scheme for `filter`:
+    - `description` - (String) The information about this filter to help identify the purpose of it.
```
Comment (geethasathyamurthy) [inline website/docs/d/cis_firewall_rules.html.markdown:36]: The information about this filter helps identify its purpose.

File: website/docs/r/cis_firewall_rules.html.markdown [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+}
+```
+
+## Argument reference
+Review the argument references that you can specify for your resource. 
+
+- `cis_id` - (Required, String) The ID of the IBM Cloud Internet Services instance where you want to create the firewall rules.
+- `domain_id` - (Required, String) The ID of the domain where you want to apply the firewall rules.
+- `filter_id` - (Required, String) The type of filter id from which you want to create firewall rules.
```
Comment (geethasathyamurthy) [inline website/docs/r/cis_firewall_rules.html.markdown:44]: id to be as ID

File: website/docs/r/cis_firewall_rules.html.markdown [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+- `action` - (Required, String) Create a firewall rules using these actions `log`, `allow`, `challenge`, `js_challenge`, `block`
```
Comment (geethasathyamurthy) [inline website/docs/r/cis_firewall_rules.html.markdown:45]: Create firewall rules by using these `log`, `allow`, `challenge`, `js_challenge`, `block` actions.
The firewall action to perform, `log` action is only available for the Enterprise plan instances.

File: website/docs/r/cis_firewall_rules.html.markdown [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+- `filter_id` - (Required, String) The type of filter id from which you want to create firewall rules.
+- `action` - (Required, String) Create a firewall rules using these actions `log`, `allow`, `challenge`, `js_challenge`, `block` 
+The firewall action to perform, "log" action is only available for enterprise plan instances.
+- `priority` - (Optional, String) Create a firewall rules with priority.
+
+
+  Nested scheme for `filter_id`:	
+  - `expression` - (Required, String) The expression of filter.
+  - `paused` - (Optional, Bool) Whether this filter is currently disabled.
+  - `description` - (Optional, String) The information about this filter to help identify the purpose of it.
```
Comment (geethasathyamurthy) [inline website/docs/r/cis_firewall_rules.html.markdown:53]: The information about the filter helps to identify its purpose.

File: website/docs/r/cis_firewall_rules.html.markdown [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+  Nested scheme for `filter_id`:	
+  - `expression` - (Required, String) The expression of filter.
+  - `paused` - (Optional, Bool) Whether this filter is currently disabled.
+  - `description` - (Optional, String) The information about this filter to help identify the purpose of it.  
+  
+## Attribute reference
+In addition to all arguments above, the following attributes are exported:
+
+- `filter_id` - (String) The filter id.
```
Comment (geethasathyamurthy) [inline website/docs/r/cis_firewall_rules.html.markdown:58]: id to be as ID

File: website/docs/r/cis_firewall_rules.html.markdown [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+**Syntax**
+
+```
+$ terraform import ibm_cis_firewall_rules.myorg <firewall_rules_id>:<domain-id>:<crn>
+```
+
+**Example**
+
+```
+$ terraform import ibm_cis_firewall_rules.myorg
```
Comment (geethasathyamurthy) [inline website/docs/r/cis_firewall_rules.html.markdown:81]: Hope the provided ID is not the actual ID details, if yes, edit the ID to dummy text.
d72c91492cc24d8286fb713d406abe91:0b30801280dc2dacac1c3960c33b9ccb:crn:v1:bluemix:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:9054ad06-3485-421a-9300-fe3fb4b79e1d::

**Review by geethasathyamurthy — COMMENTED** [review]
Reviewed the docs update, requests for incorporating the comments.

File: website/docs/r/cis_firewall_rules.html.markdown (hunk 7 above)
Comment (kavya498) [reply website/docs/r/cis_firewall_rules.html.markdown:44]: @geethasathyamurthy , `cis_id` schema argument is `cis_id`..
It should not be changed

File: website/docs/r/cis_firewall_rules.html.markdown (hunk 8 above)
Comment (shivaleelac) [reply website/docs/r/cis_firewall_rules.html.markdown:44]: modified as above

File: website/docs/r/cis_firewall_rules.html.markdown (hunk 11 above)
Comment (shivaleelac) [reply website/docs/r/cis_firewall_rules.html.markdown]: Its dummy ID

File: website/docs/r/cis_firewall_rules.html.markdown (hunk 10 above)
Comment (shivaleelac) [reply website/docs/r/cis_firewall_rules.html.markdown:55]: changed it as ID

File: website/docs/r/cis_firewall_rules.html.markdown (hunk 9 above)
Comment (shivaleelac) [reply website/docs/r/cis_firewall_rules.html.markdown]: Changed

File: website/docs/d/cis_firewall_rules.html.markdown (hunk 6 above)
Comment (shivaleelac) [reply website/docs/d/cis_firewall_rules.html.markdown]: changed

File: website/docs/d/cis_firewall_rules.html.markdown [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+## Argument reference
+The following arguments are supported:
+
+- `cis_id` - (Required, String) The ID of the CIS service instance.
+- `domain_id` - (Required, String) The ID of the domain.
+
+## Attributes reference
+In addition to all arguments above, the following attributes are exported:
+
+- `firewall_rules` - (List of Firewall Rules)
```
Comment (geethasathyamurthy) [inline website/docs/d/cis_firewall_rules.html.markdown:30]: Provide 
Nested schema for `firewall_rules`:

File: website/docs/r/cis_firewall_rules.html.markdown [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+subcategory: "Internet services"
+layout: "ibm"
+page_title: "IBM: ibm_cis_firewall_rules"
+description: |-
+  Provides a IBM CIS Firewall Rules resource.
+---
+
+# ibm_cis_firewall_rules
+
+
```
Comment (geethasathyamurthy) [inline website/docs/r/cis_firewall_rules.html.markdown:12]: Delete a line here

File: website/docs/d/cis_firewall_rules.html.markdown (hunk 12 above)
Comment (shivaleelac) [reply website/docs/d/cis_firewall_rules.html.markdown:30]: fixed

**Review by geethasathyamurthy — APPROVED** [review]
Docs comments are fixed.

**Comment by hkantare** [comment]
Can you please fix the conflicts

**Comment by kavya498** [comment]
Closing this as it is already merged cia other PR

**Pull request closed without merging**